src/app/features/section/subpages/edit-form/form/form.component.ts

```diff
@@ -0,0 +1,176 @@
+import { MatAutocompleteModule, MatAutocompleteSelectedEvent } from '@angular/material/autocomplete';
+import { MatButtonModule } from '@angular/material/button';
+import { MatChipsModule } from '@angular/material/chips';
+import { MatFormFieldModule } from '@angular/material/form-field';
+import { MatIconModule } from '@angular/material/icon';
+import { MatInputModule } from '@angular/material/input';
+import { ActivatedRoute } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Observable, map, switchMap, of } from 'rxjs';
+
+import { User, selectSectionDetails, selectAllUsers, SectionActions, sectionDetailsActions } from '../../../';
+import { validateCharacters, validateNotNumbers } from '../';
+
+@Component({
+  selector: 'app-form',
+  standalone: true,
+  imports: [
+    MatIconModule,
+    MatFormFieldModule,
+    AsyncPipe,
+    MatInputModule,
+    MatIconModule,
+    MatButtonModule,
+    NgIf,
+    NgForOf,
+    ReactiveFormsModule,
+    FormsModule,
+    MatChipsModule,
+    MatAutocompleteModule,
+  ],
+  templateUrl: './form.component.html',
+  styleUrls: ['./form.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class FormComponent implements OnInit {
+  private formBuilder = inject(NonNullableFormBuilder);
+  private activeRoute = inject(ActivatedRoute);
+  private store = inject(Store);
+
+  sectionDetails$ = this.store.select(selectSectionDetails);
+  allUsers$ = this.store.select(selectAllUsers);
+  editSectionForm = this.editSectionsForm();
+  filteredUsers$!: Observable<User[]>;
+  editedOwner!: User;
+
+  ngOnInit() {
+    this.activeRoute.parent?.paramMap
+      .pipe(
+        map(params => params?.get('id')),
+        switchMap(id =>
+          id ? of(this.store.dispatch(sectionDetailsActions.getSectionDetails({ sectionId: +id }))) : of([])
+        )
+      )
+      .subscribe();
+
+    this.sectionDetails$.subscribe(section => {
+      this.editSectionForm.patchValue(section);
+      this.editedOwner = section.sectionOwner;
+    });
+  }
+
+  selectedUser(event: MatAutocompleteSelectedEvent) {
+    this.editedOwner = event.option.value;
+  }
+
+  getFilteredUsers() {
+    this.filteredUsers$ = this.allUsers$.pipe(
+      map(users =>
+        users.filter(user => {
+          const isSelected = user.id !== this.editedOwner.id;
+
+          if (!isSelected) {
+            return isSelected;
+          } else {
+            const isIncluded = (user.firstName + ' ' + user.lastName)
+              .toLowerCase()
+              .includes(String(this.editSectionForm.controls.sectionOwner.getRawValue()).toLowerCase());
+
+            return isIncluded;
+          }
+        })
+      )
+    );
+  }
+
+  getErrorMessage(formControlName: 'name' | 'description') {
+    const control = this.editSectionForm.get(formControlName);
+    const errorsTree = {
+      name: {
+        required: 'Pole wymagane',
```

Comment:
ogólnie wiele komunikatów się powtarza, myślę że warto dodać do backlogu task by to uprządkować, stworzyć jakiś globalny słownik błędów i korzystać z niego